### DISCUSSION TITLE
Automatic build workflow

### DISCUSSION THREAD

**Discussion by kiraio-moe** [Tualin14/baPlayer#24] [Development]
Hi @Tualin14, can you activate the Unity license key generated by [Get Unity License](https://github.com/Tualin14/ArisStudio/actions/workflows/get_unity_license.yml) action? I can't sent the activated license file in here. Plus add a secret key to the repo, I don't have the permission to do this.

I'm following an article on how to build a unity project with GitHub workflow. Here's the article: https://blog.yarsalabs.com/automating-builds-in-unity-github-gameci/

I have been setting up the actions and the rest is:
- Download the [Manual Activation File](https://github.com/Tualin14/ArisStudio/suites/12921562581/artifacts/697965257) and save as `.zip`. You can see it at [Get Unity License > Summary > Artifacts section](https://github.com/Tualin14/ArisStudio/actions/runs/4988552010).
- Extract and activate it at [Unity Manual Activation](https://license.unity3d.com/manual).
- Don't forget to keep safe the `.alf` & `.ulf` files.
- Head to repo's Secret setting, create a new secret by the name `UNITY_LICENSE` and the token is the content of `.ulf` file.
- Done. See [Action](https://github.com/Tualin14/ArisStudio/actions) tab to make sure `Build Aris Studio` action is running.

**Comment by ttttttttttttttllllllllllllll** [comment]
Sorry for the late reply, I don't have my computer with me and I will be able to add it tomorrow night.

**Comment by kiraio-moe** [comment]
No worry, it's not that important.

Pada tanggal Sel, 16 Mei 2023 19.13, dzaaaaaa ***@***.***>
menulis:

**Reply by ttttttttttttttllllllllllllll** [reply]
`serial has reached the maximum number of activations.`

Maybe it's because I'm a Unity Personal Edition ~(>_<。)＼

**Reply by kiraio-moe** [reply]
Will try it on my self then, if luck.

**Reply by ttttttttttttttllllllllllllll** [reply]
UNITY_LICENSE Added
![image](https://github.com/Tualin14/ArisStudio/assets/71857734/d18c03bd-3d70-4d38-9832-191af763f406)

**Reply by kiraio-moe** [reply]
Thanks for the job 🙏

**Reply by kiraio-moe** [reply]
The automatic build failing because we didn't have Spine in our repository: https://github.com/Tualin14/ArisStudio/actions/runs/5004239973/jobs/8967208074#step:7:4011

```log
-----CompilerOutput:-stdout--exitcode: 1--compilationhadfailure: True--outfile: Temp/ArisStudio.dll
Microsoft (R) Visual C# Compiler version 3.5.0-dev-20359-01 (8da8ba0c)
Copyright (C) Microsoft Corporation. All rights reserved.

Assets/ArisStudio/AsGameObject/AsCharacterManager.cs(7,7): error CS0246: The type or namespace name 'Spine' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/AsGameObject/AsCharacterManager.cs(8,7): error CS0246: The type or namespace name 'Spine' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/AsGameObject/Character/AsCharacter.cs(3,7): error CS0246: The type or namespace name 'Spine' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/StatePreview/PreSpr.cs(5,7): error CS0246: The type or namespace name 'Spine' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/StatePreview/PreSpr.cs(6,7): error CS0246: The type or namespace name 'Spine' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/Utils/SpineHelper.cs(3,7): error CS0246: The type or namespace name 'Spine' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/Utils/SpineHelper.cs(4,7): error CS0246: The type or namespace name 'Spine' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/StatePreview/PreSpr.cs(26,17): error CS0246: The type or namespace name 'SkeletonAnimation' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/Utils/SpineHelper.cs(11,63): error CS0246: The type or namespace name 'SkeletonData' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/Utils/SpineHelper.cs(11,90): error CS0246: The type or namespace name 'AnimationStateData' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/Utils/SpineHelper.cs(11,23): error CS0246: The type or namespace name 'SkeletonDataAsset' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/AsGameObject/Character/AsCharacter.cs(12,17): error CS0246: The type or namespace name 'SkeletonAnimation' could not be found (are you missing a using directive or an assembly reference?)
-----EndCompilerOutput---------------
- Finished compile Library/ScriptAssemblies/Unity.Mathematics.Editor.dll in 1.217553 seconds
- Finished script compilation in 34.284518 seconds
Assets/ArisStudio/AsGameObject/AsCharacterManager.cs(7,7): error CS0246: The type or namespace name 'Spine' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/AsGameObject/AsCharacterManager.cs(8,7): error CS0246: The type or namespace name 'Spine' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/AsGameObject/Character/AsCharacter.cs(3,7): error CS0246: The type or namespace name 'Spine' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/StatePreview/PreSpr.cs(5,7): error CS0246: The type or namespace name 'Spine' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/StatePreview/PreSpr.cs(6,7): error CS0246: The type or namespace name 'Spine' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/Utils/SpineHelper.cs(3,7): error CS0246: The type or namespace name 'Spine' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/Utils/SpineHelper.cs(4,7): error CS0246: The type or namespace name 'Spine' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/StatePreview/PreSpr.cs(26,17): error CS0246: The type or namespace name 'SkeletonAnimation' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/Utils/SpineHelper.cs(11,63): error CS0246: The type or namespace name 'SkeletonData' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/Utils/SpineHelper.cs(11,90): error CS0246: The type or namespace name 'AnimationStateData' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/Utils/SpineHelper.cs(11,23): error CS0246: The type or namespace name 'SkeletonDataAsset' could not be found (are you missing a using directive or an assembly reference?)
Assets/ArisStudio/AsGameObject/Character/AsCharacter.cs(12,17): error CS0246: The type or namespace name 'SkeletonAnimation' could not be found (are you missing a using directive or an assembly reference?)
AssetDatabase: script compilation time: 36.315497s
Scripts have compiler errors.
```

**Reply by ttttttttttttttllllllllllllll** [reply]
The article says
`The generated file contains your secret Unity license data. Make sure you don't share or disclose them with anyone else.`

So I deleted the download link you shared above.

**Reply by ttttttttttttttllllllllllllll** [reply]
> 自动构建失败，因为我们的存储库中没有Spine：https://github.com/Tualin14/ArisStudio/actions/runs/5004239973/jobs/8967208074#step:7:4011

We can add clone spine repo in workflow
https://github.com/EsotericSoftware/spine-runtimes/tree/3.8/spine-unity/Assets/Spine

**Reply by kiraio-moe** [reply]
Why did spine-unity didn't recognize spine-csharp even though they're in the same assembly? weird... 🤔 

```log
...
Assets/Spine/Runtime/spine-unity/Utility/SkeletonExtensions.cs(408,22): error CS0246: The type or namespace name 'BoneData' could not be found (are you missing a using directive or an assembly reference?)
Assets/Spine/Runtime/spine-unity/Utility/SkeletonExtensions.cs(421,22): error CS0246: The type or namespace name 'Bone' could not be found (are you missing a using directive or an assembly reference?)
Assets/Spine/Runtime/spine-unity/Utility/SkeletonExtensions.cs(446,39): error CS0246: The type or namespace name 'VertexAttachment' could not be found (are you missing a using directive or an assembly reference?)
Assets/Spine/Runtime/spine-unity/Utility/SkeletonExtensions.cs(450,41): error CS0246: The type or namespace name 'Attachment' could not be found (are you missing a using directive or an assembly reference?)
...
```

**Reply by kiraio-moe** [reply]
> Why did spine-unity didn't recognize spine-csharp even though they're in the same assembly? weird... 🤔
> 
> ```
> ...
> Assets/Spine/Runtime/spine-unity/Utility/SkeletonExtensions.cs(408,22): error CS0246: The type or namespace name 'BoneData' could not be found (are you missing a using directive or an assembly reference?)
> Assets/Spine/Runtime/spine-unity/Utility/SkeletonExtensions.cs(421,22): error CS0246: The type or namespace name 'Bone' could not be found (are you missing a using directive or an assembly reference?)
> Assets/Spine/Runtime/spine-unity/Utility/SkeletonExtensions.cs(446,39): error CS0246: The type or namespace name 'VertexAttachment' could not be found (are you missing a using directive or an assembly reference?)
> Assets/Spine/Runtime/spine-unity/Utility/SkeletonExtensions.cs(450,41): error CS0246: The type or namespace name 'Attachment' could not be found (are you missing a using directive or an assembly reference?)
> ...
> ```

Wait, bruh... Why did `svn` cloning only the `spine-unity`?
https://github.com/Tualin14/ArisStudio/actions/runs/5009948829/jobs/8979382818#step:3:205

```log
A    Assets/Spine/Runtime.meta
A    Assets/Spine/Runtime
A    Assets/Spine/Runtime/spine-csharp.meta
A    Assets/Spine/Runtime/spine-csharp
A    Assets/Spine/Runtime/spine-csharp/add spine-csharp here.txt
A    Assets/Spine/Runtime/spine-unity.asmdef
A    Assets/Spine/Runtime/spine-unity.asmdef.meta
A    Assets/Spine/Runtime/spine-unity.meta
A    Assets/Spine/Runtime/spine-unity
```

**Reply by ttttttttttttttllllllllllllll** [reply]
Oh! I didn't notice https://github.com/EsotericSoftware/spine-runtimes/tree/3.8/spine-unity/Assets/Spine/Runtime/spine-csharp is empty

**Reply by kiraio-moe** [reply]
> Oh! I didn't notice https://github.com/EsotericSoftware/spine-runtimes/tree/3.8/spine-unity/Assets/Spine/Runtime/spine-csharp is empty

I didn't realize it 🤣

**Comment by kiraio-moe** [comment]
@Tualin, I think you should rebase the commits from 16 - 18 may, it's a mess right now.

**Reply by ttttttttttttttllllllllllllll** [reply]
complete

**Comment by ttttttttttttttllllllllllllll** [comment]
It's work https://github.com/Tualin14/ArisStudio/actions/runs/5010587459

**Reply by kiraio-moe** [reply]
Yeah 🎉
